Fix markup background, improve wiki rendering

silverwind opened the pull request:
Fix regression from https://github.com/go-gitea/gitea/pull/23578. Only visible on arc-green.

Before:
<img width="997" alt="Screenshot 2023-03-27 at 19 14 21" src="https://user-images.githubusercontent.com/115237/228016589-e7cabfb9-bfd0-45fd-9407-6b76c665ed1a.png">

After:
<img width="1000" alt="Screenshot 2023-03-27 at 19 14 05" src="https://user-images.githubusercontent.com/115237/228016600-db2e6002-4e2c-4d18-8393-9d7e1f525acb.png">

Fixes: https://github.com/go-gitea/gitea/issues/20625
Fixes: https://github.com/go-gitea/gitea/issues/23718

wxiaoguang commented:
I think we do not need this PR. Instead, we could fix the use `background: inherit` for the markup? Or just remove it?

https://github.com/go-gitea/gitea/pull/23578/files#r1149604909

silverwind commented:
If we ensure the parent elements set the background, we can indeed remove the rule from `.markup`, making this removal unnecessary.

silverwind commented:
There are many places where we allow markdown that need exclusions. I think I'm leaning towards only supporting these opaque images in a few specific places like file view, comments and wiki.

silverwind commented:
Done, it now only sets opaque on areas where we know the background beforehand, which is:

- File view including README
- Issue comments, both regular and code
- Wiki content

I also fixed another related regression I noticed on the comment edit page where I added padding:

<img width="853" alt="image" src="https://user-images.githubusercontent.com/115237/228083833-94fa91c5-fb5f-4cc3-897f-ea5836e9319e.png">

(btw easymde looks horrible there 😆)

wxiaoguang commented:
I like this new approach. At least  we  do not need to overwrite `.markup { background }` anymore.  Thank you ~~~

wxiaoguang commented:
According to #23578, I do not think we should do `.comment-body { padding }` +  `.repository .comment.form { padding: 1em;}`, it seems a overwriting whack-a-mole game again.

The clearest solution in my mind is:

* Keep `.comment-body` as before.
* Only set `.comment-body .markup { padding: 0 }`

The philosophy is that every element should declare its own styles clearly, but do not hack the general styles.

silverwind commented:
Moved the paddings to parent elements, except wiki which still has it on the same element because it's not possible without an additional wrapper `<div>`. Other than that, this includes a few more small fixes:

Improved wiki padding so header link anchor is clickable:

<img width="1238" alt="Screenshot 2023-04-02 at 23 59 48" src="https://user-images.githubusercontent.com/115237/229381539-9fbe1ecc-0416-4a4b-a95a-daea3899edef.png">

Fixed pages list right side background:

<img width="1243" alt="Screenshot 2023-04-02 at 23 59 19" src="https://user-images.githubusercontent.com/115237/229381587-603fa2c6-671e-4a8e-b9bf-e2c388b1b2d6.png">

Improved alignment of text and vertical centering of avatar in inline comments:

<img width="416" alt="Screenshot 2023-04-02 at 23 45 46" src="https://user-images.githubusercontent.com/115237/229381599-a4b0cf3f-c1cf-4dd8-bb65-7171a70823eb.png">
<img width="827" alt="Screenshot 2023-04-03 at 00 09 38" src="https://user-images.githubusercontent.com/115237/229381640-49b8d456-429f-46bf-8b51-9eedf626cfb7.png">

silverwind commented on web_src/css/repository.css:
```diff
@@ -2120,17 +2124,13 @@
-.repository.wiki.view > .markup h4:first-of-type,
-.repository.wiki.view > .markup h5:first-of-type,
-.repository.wiki.view > .markup h6:first-of-type {
-  margin-top: 0;
```
These are dead code, `.markup` is no direct child of `.view` probably since a very long time.

silverwind commented on web_src/css/repository.css:
```diff
@@ -2761,11 +2761,11 @@
 
 .comment-body .markup {
-  padding: 1em;
+
```
```suggestion
```

jolheiser commented:
🎺 🤖
